pkg/operator/controller/controller.go

```diff
@@ -259,20 +257,27 @@ func (r *ReconcileCDI) reconcileUpdate(logger logr.Logger, cr *cdiv1alpha1.CDI)
 		logger.Info("Successfully entered Deployed state")
 	}
 
-	if err = r.checkReady(logger, cr); err != nil {
+	ready, err := r.checkReady(logger, cr)
+	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	if ready {
+		if err = r.ensureUploadProxyRouteExists(logger, cr); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileCDI) shouldUpdateObject(obj runtime.Object) bool {
+func (r *ReconcileCDI) isMutable(obj runtime.Object) bool {
 	switch obj.(type) {
 	case *corev1.Secret:
```

Comment:
So golang doesn't by default fall through to the next statement like a lot of other languages, so the secret will return false, and the configmap will return true, but its a little confusing from the code to know which one you intended, because I would leave Secret out of the switch statement if I wanted to return false.

Reply:
Oh man, great catch!  Clearly I was confused.  Fortunately, the operator doesn't create any secrets at this point so wasn't an issue.  But will remove the secret case for clarity.